### PR TITLE
feat(#42): Setup mongdb with docker

### LINKED ISSUE #42: Setup MongoDB with Docker
MongoDB enables high availability and scalability. It works well in distributed environments like Docker containers. Using MongoDB with Docker allows us to have a portable database that can be run on any server platform without having to worry about its configuration. We can use Docker with a MongoDB container image to make the database deployment process more efficient and straightforward.

Task:
- Add MongoDB to the list of existing containers
- Make Database accessible to Restful services
- Port mapping
- Volume mount data

### PR DESCRIPTION
**Pull request by tholulomo**
close #42

### CODE REVIEW

File: resfulservice/misc/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+On the terminal:
+1. Go to => `~/.bash_profile` (If not found create one)
+2. Scroll down to the end of the `.bash_profile` file
+3. Download gpg [here](https://gpgtools.org/) and request decryption keys for `envVarFile.gpg`
+4. Decrypt `envVarFile.gpg` content by executing `decryptENVARFile.sh` in this directory
+5. Copy and paste `envVarFile` content
```
Comment (stouffers) [inline resfulservice/misc/README.md:6]: It may make more sense to the reader to list these three steps before creating / opening up .bash_profile

**Review by stouffers — COMMENTED** [review]
Have a few comments but generally looking good!

File: mockDB/mongodb/.dbshell [hunk 2]
```diff
@@ -0,0 +1,5 @@
+var rootUser = 'MONGODEVOWNER';
+var rootPassword = 'nM4455BXWy';
+var admin = db.getSiblingDB('admin');
+var user = '';
+var passwd = '>LQeenKey845$';
```
[the remark this answers is not in the document]
Comment (tholulomo) [reply mockDB/mongodb/.dbshell:5]: @stouffers, I made a mistake uploading this. I believed I changed this keys, I'll update

File: docker-compose.yml [hunk 3]
```diff
@@ -1,5 +1,21 @@
 version: '3'
 services:
+  mongo:
+    image: 'mongo:latest'
```
[the remark this answers is not in the document]
Comment (tholulomo) [reply docker-compose.yml:4]: @stouffers, this composer is for dev purposes and not prod. But I'll pin it to a major version as suggested.

File: resfulservice/misc/README.md (hunk 1 above)
Comment (tholulomo) [reply resfulservice/misc/README.md:6]: @stouffers, thanks for this suggestion. I'll update.

File: docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tholulomo) [reply docker-compose.yml:4]: I just pinned it to 4.0.4, @stouffers do you have any version in mind?

[the remark this answers is not in the document]
Comment (stouffers) [reply docker-compose.yml:4]: Not familiar enough with Mongo to recommend a specific version.  4.0 seems old though.  It looks like 5 has been released recently, though 4.4 may be a safer bet.

[the remark this answers is not in the document]
Comment (tholulomo) [reply docker-compose.yml:4]: Thanks, I will also look into it. In the meantime, I will use 4.4

**Review by aswallace — APPROVED** [review]

File: resfulservice/misc/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+1. Download gpg [here](https://gpgtools.org/) and request decryption keys for `envVarFile.gpg`
+
+On the terminal:
+2. Decrypt `envVarFile.gpg` content by executing `decryptENVARFile.sh` in this directory
+3. Go to => `~/.bash_profile` (If not found create one)
+4. Scroll down to the end of the `.bash_profile` file
+5. Copy and paste `envVarFile` content
+6. Save any changes you made to the `.bash_profile` file
+7. Execute the new .bash_profile by running => `source ~/.bash-profile`
```
Comment (stouffers) [inline resfulservice/misc/README.md:9]: This doesn't appear to render as intended

https://github.com/Duke-MatSci/materialsmine/blob/952115524b3c52516639d28472075b8d09ee4905/resfulservice/misc/README.md

Comment (tholulomo) [reply resfulservice/misc/README.md:9]: 😂 . I'll take a look, nice catch

**Review by stouffers — APPROVED** [review]

**Pull request merged**